### ISSUE TITLE
Feature:  Auto-Hide

### ISSUE THREAD

**Issue by erffy** [Alexays/Waybar#3853]
There is a 'Hide' feature but it is not as desired, the bar can disappear from the top and when we move the mouse to the part of the layout where it is set, it becomes visible and when we move it away it hides back, so we gain more space on the screen.

**Comment by UnaTried** [comment]
You can set the size to 5px in css and to 50px (or your size) on hover.

**Comment by zmeyka3310** [comment]
bump, i really need this. also @UnaTried could you please provide a config example?

**Comment by unatried** [comment]
@zmeyka3310 
Options 1:
```css
window#waybar {
    min-height: 5px;
}
window#waybar:hover {
    min-height: 50px;
}
```

Options 2:
```jsonc
{
    "modifier-reset": "press", // or "release" if you want it on release of the key.
}
```

Options 3:
```jsonc
{
    "mode": "invisible", // or "hidden", then you should use the sway-bar mode setting https://github.com/swaywm/sway/blob/master/sway/sway-bar.5.scd}
}
```
I don't know how to set the modifier key though

**Comment by inevity** [comment]
@UnaTried where set the option 1?

**Comment by unatried** [comment]
@inevity in your style.css or whatever .css file you use to style waybar...

**Comment by Invertisment** [comment]
This was merged yesterday: https://github.com/Alexays/Waybar/issues/3928.You can have an external script to control the bar hiding using signals.

**Comment by GG2R10** [comment]
> This was merged yesterday: [#3928](https://github.com/Alexays/Waybar/issues/3928) can have an external script to control the bar hiding using signals.

So, I need to have a waybar instance for each of my bars right? I have a 3 bar setup in the same configuration file and waybar only has one process because of this. I'm a little confused becuase even the man page says "per bar", but in practice you can't send a SIGUSR signal to a specific bar if there are multiple created in the same config file / instance of waybar :x

**Comment by Invertisment** [comment]
The reaction is per-bar. If you want to target a specific bar then you have to run them so they would have their own PIDs.
And then use those PIDs for targeting.
I don't know if it's bad but at least now there is a way for controlling the bars.

What I currently do is match the `pgrep -a waybar` and target the PID.

**Comment by unatried** [comment]
@Invertisment what you are explaining is exactly what @GG2R10 said. 

@GG2R10, the answer is: Make multiple config files for multiple waybar instances, so you can use the new feature.

**Comment by GG2R10** [comment]
> The reaction is per-bar. If you want to target a specific bar then you have to run them so they would have their own PIDs. And then use those PIDs for targeting. I don't know if it's bad but at least now there is a way for controlling the bars.
> 
> What I currently do is match the `pgrep -a waybar` and target the PID.

No, it is not bad and ty for implementing this functionality in waybar. I had to change some configurations but now im using it with a toggle bind and it's working nice! The only thing i haven't figured is how to animate the hidding / unhidding of the bars

**Comment by unatried** [comment]
> The only thing i haven't figured is how to animate the hidding / unhidding of the bars

**I cannot use my PC right now, so I haven't looked into the feature**

Well since it kills the bar with a signal, I don't think you can animate that. Maybe there is a pre sending signal hook, I don't know, but if that doesn't exist I don't think you could.

I believe Hyprland does have a layer exit animation. (Because waybar is a layer)

**Comment by Invertisment** [comment]
A regular kill signal will kill it.

But if you send `kill -SIGUSR1 <pid>` then it will send the `SIGUSR1` signal, which you can configure in your config. There are two of these available.
See the docs.

**Comment by Zephirus2** [comment]
I made an small utility that basically does that if you want. It issues kill -SIGUSR1 and kill -SIGUSR2 to turn it on and off 

Here's the link, it's lightweight and you just have to add it to your exec-once in your hyprland config.
https://github.com/Zephirus2/waybar_auto_hide

**Comment by unatried** [comment]
It's already merged...

**Comment by cipher-iso** [comment]
Just if anyone is still looking, I made an easy script for this with [instructions](https://github.com/cipher-iso/Waybar-Autohide)

**Comment by Alexays** [comment]
Closing — this was resolved/answered in the thread (config, environment, or upstream — not a Waybar bug still open). Feel free to reopen if you're still stuck. 🙏
